pkg/api/vm/formatter.go

```diff
@@ -2,36 +2,61 @@ package vm
 
 import (
 	"github.com/rancher/apiserver/pkg/types"
+	ctlkubevirtv1alpha3 "github.com/rancher/harvester/pkg/generated/controllers/kubevirt.io/v1alpha3"
 	"github.com/rancher/steve/pkg/resources/common"
 	"github.com/rancher/wrangler/pkg/data/convert"
 
 	v1alpha3 "kubevirt.io/client-go/api/v1alpha3"
 )
 
 const (
-	startVM    = "startVM"
-	stopVM     = "stopVM"
-	restartVM  = "restartVM"
+	startVM    = "start"
+	stopVM     = "stop"
+	restartVM  = "restart"
+	pauseVM    = "pause"
+	unpauseVM  = "unpause"
 	ejectCdRom = "ejectCdRom"
 )
 
-func formatter(request *types.APIRequest, resource *types.RawResource) {
+type vmformatter struct {
+	vmiCache ctlkubevirtv1alpha3.VirtualMachineInstanceCache
+}
+
+func (vf *vmformatter) formatter(request *types.APIRequest, resource *types.RawResource) {
 	common.Formatter(request, resource)
 	// reset resource actions, because action map already be set when add actions handler,
 	// but current framework can't support use formatter to remove key from action map
 	resource.Actions = make(map[string]string, 1)
-	if resource.APIObject.Data().Bool("spec", "running") {
+
+	var vm v1alpha3.VirtualMachine
```

Comment:
We are mostly using its reference in the following codes. Can we define a pointer here?